src/api/app/controllers/webui/users/notifications_controller.rb

```diff
@@ -51,21 +46,23 @@ def update
   private
 
   def set_filter_kind
-    @filter_kind = Array(params[:kind].presence || 'all') # in case just one value, we want an array anyway
-    raise FilterNotSupportedError unless (@filter_kind - ALLOWED_FILTERS).empty?
+    @filter_kind = Array(params[:kind].presence || 'all')
+    @filter_kind.reject! { |kind| ALLOWED_FILTERS.exclude?(kind) }
   end
 
   def set_filter_state
     @filter_state = params[:state].presence || 'unread'
-    raise FilterNotSupportedError if ALLOWED_STATES.exclude?(@filter_state)
+    @filter_state = 'unread' if %w[read unread].exclude?(@filter_state)
```

Comment:
```suggestion
    @filter_state = 'unread' if ALLOWED_STATES.exclude?(@filter_state)
```

I think you should still use the `ALLOWED_STATES` which includes the `all` possible value too.